abpframework/abp
# CmsKit - PreventXSS Enhancement

### Description

Resolves https://github.com/abpframework/abp/issues/17677

**PreventXSS** feature was enabled by default for all contents. It's a problem while adding javascript to the CMS Application and trying to call any function from that javascript.

Now, it doesn't work everywhere by default. It'll work on the following conditions after this PR:

- **Pages**: Disabled by default.
- **Blogs**: Enabled by default. (Configurable per Blog)
- **Comments**: Enabled by default

![preventxss](https://github.com/abpframework/abp/assets/23705418/bd0b12c5-99a6-4ff4-9e27-cc7b3c6a756c)

### Checklist


### How to test it?

1. Add the following script by using **Global Resources**  
  ```js
  function sayHello(){
    alert('Hello');
  }
```

2. Create a Page or BlogPost with the following button

  ```html
   <button class="btn btn-primary" onclick="sayHello()">Click me</button>
  ```

Page should work by default, BlogPost shouldn't work until you disable PreventXSS feature for blog.
